### PR TITLE
fix(windows): Windows compatibility for azure-devops, clarity & shared utils

### PR DESCRIPTION
**Pull request by genesiscz**
- Guard `chmod(0o600)` behind `process.platform !== "win32"` in clarity config
- Platform-aware error messages in Storage (suggests `notepad`/`copy` on Windows instead of `$EDITOR`/`cp`)
- Cross-platform `escapeShellArg` — double-quote escaping for `cmd.exe` on Windows
- Replace `!path.startsWith("/")` with `isAbsolute()` in file-path prompt utility
- Handle `\` path separators throughout file-path input (tab completion, dir listing)
- Windows-appropriate GitHub CLI config path (`%APPDATA%\gh\` instead of `~/.config/gh/`)
- Handle `EPERM` in file-lock process-alive check (Windows returns EPERM for existing processes)
> **Reviewers:** Thoroughly check for Windows incompatibilities in all tools this PR touches.
Key areas to verify:
- `src/utils/prompts/clack/file-path.ts` — most changes; `endsWithSep`/`lastSepIndex` helpers replace all hardcoded `/` checks
- `src/utils/string.ts` — `escapeShellArg` now branches on `process.platform`
- `src/utils/storage/file-lock.ts` — EPERM handling in `isProcessAlive`
- Bun `$` shell calls in `src/azure-devops/` were analyzed and deemed safe (Bun Shell is cross-platform) — no changes needed there
- [x] `tsgo --noEmit` — zero type errors
- [x] `bun test src/utils/string.test.ts` — 41 tests pass
- [ ] Manual: run `tools azure-devops --help` and `tools clarity --help` on Windows with Bun
- [ ] Verify `escapeShellArg` produces valid `cmd.exe` quoting on Windows
- [ ] Verify file-path prompt handles `C:\Users\...` style paths correctly

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved Windows compatibility for saving config, GitHub token lookup, process liveness checks, shell-argument quoting, and daemon detection
  * Platform-aware error messages and safer profiler behavior on non-macOS

* **New Features**
  * Cross-platform path utilities and separator-aware file-path input and glob handling
  * Unified tilde/relative-path expansion used across prompts and watchers
  * More robust audio segment naming across path formats

* **Tests**
  * Added cross-platform tests for path utilities and Windows/Unix shell-escaping

### CODE REVIEW

File: src/utils/string.ts [hunk 1]
```diff
@@ -26,9 +26,14 @@ export function stripAnsi(input: string): string {
...
 
 /**
- * Escape a string for safe use as a shell argument (single-quoted).
+ * Escape a string for safe use as a shell argument.
+ * Uses single-quoting on Unix, double-quoting on Windows cmd.exe.
  */
 export function escapeShellArg(arg: string): string {
+    if (process.platform === "win32") {
+        return `"${arg.replace(/"/g, '\\"')}"`;
+    }
```
[the remark this answers is not in the document]
Comment (genesiscz) [reply src/utils/string.ts:52]: @coderabbitai Fixed in 6388c874 — escapeShellArg now implements proper CommandLineToArgvW escaping (backslashes before quotes doubled, trailing backslashes doubled) and neutralizes cmd.exe %VAR% expansion by doubling %. Added comprehensive Windows-specific tests covering trailing backslashes, backslash-before-quote, percent expansion, and combined edge cases. Note: the &|<> metacharacters are already safe inside double quotes in cmd.exe.

File: src/utils/string.ts [hunk 2]
```diff
@@ -26,9 +26,14 @@ export function stripAnsi(input: string): string {
 }
 
 /**
- * Escape a string for safe use as a shell argument (single-quoted).
+ * Escape a string for safe use as a shell argument.
+ * Uses single-quoting on Unix, double-quoting on Windows cmd.exe.
  */
 export function escapeShellArg(arg: string): string {
+    if (process.platform === "win32") {
+        return `"${arg.replace(/"/g, '\\"')}"`;
```
[the remark this answers is not in the document]
Comment (genesiscz) [reply src/utils/string.ts]: @gemini-code-assist Fixed in 6388c874 — implemented proper CommandLineToArgvW-compatible escaping using the pattern from the spec: backslashes before a double-quote are doubled, trailing backslashes are doubled, and double-quotes are escaped. Also added cmd.exe percent neutralization. Used the more precise regex approach rather than escaping all backslashes indiscriminately.

[the remark this answers is not in the document]
Comment (genesiscz) [reply src/utils/string.ts]: @gemini-code-assist The current escapeShellArg implementation already handles both Windows (CommandLineToArgvW + cmd.exe metacharacters, modeled after cross-spawn v7.0.5+) and Unix (single-quote escaping). No changes needed — this is working as designed.

File: src/utils/paths.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
+/**
+ * Cross-platform path helpers.
+ * Handles tilde expansion, separator detection, and path resolution
+ * that work correctly on both Unix and Windows.
+ */
+
+import { homedir } from "node:os";
+import { isAbsolute, join, sep } from "node:path";
```
[the remark this answers is not in the document]
Comment (genesiscz) [reply src/utils/paths.ts]: @coderabbitai Fixed — expandPath now uses resolve() instead of join() for reliable absolute path conversion. [018ad30](https://github.com/genesiscz/GenesisTools/commit/018ad30e).

File: src/utils/paths.test.ts [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+        const { escapeShellArg } = await import("../utils/string");
+        expect(escapeShellArg("foo & bar")).toBe('^"foo^ ^&^ bar^"');
+    });
+
+    it("uses single quotes on Unix (default)", async () => {
+        restorePlatform();
+        const { escapeShellArg } = await import("../utils/string");
+        expect(escapeShellArg("hello")).toBe("'hello'");
+    });
+});
```
[the remark this answers is not in the document]
Comment (genesiscz) [reply src/utils/paths.test.ts:255]: @coderabbitai The referenced test file paths.test.ts and string.test.ts are separate — paths.test.ts tests path utilities, string.test.ts tests string utilities including escapeShellArg. No duplication in current code.

File: src/automate/lib/daemon.ts [hunk 5]
```diff
@@ -42,7 +42,12 @@ export function getDaemonPid(): number | null {
         try {
             process.kill(pid, 0);
             return pid;
-        } catch {
+        } catch (err) {
+            // On Windows, EPERM means the process exists but we can't signal it
+            if (process.platform === "win32" && err instanceof Error && "code" in err) {
```
[the remark this answers is not in the document]
Comment (genesiscz) [reply src/automate/lib/daemon.ts]: @coderabbitai The referenced Windows EPERM comments do not exist in the current code — both daemon files use plain try/catch without platform-specific commentary. No changes needed.

File: src/daemon/daemon.ts [hunk 6]
```diff
@@ -45,7 +45,12 @@ export function getDaemonPid(): number | null {
         try {
             process.kill(pid, 0);
             return pid;
-        } catch {
+        } catch (err) {
+            // On Windows, EPERM means the process exists but we can't signal it
+            if (process.platform === "win32" && err instanceof Error && "code" in err) {
```
[the remark this answers is not in the document]
Comment (genesiscz) [reply src/daemon/daemon.ts]: @coderabbitai The referenced Windows EPERM comments do not exist in the current code — both daemon files use plain try/catch without platform-specific commentary. No changes needed.

File: src/fsevents-profile/index.ts [hunk 7]
```diff
@@ -9,6 +9,11 @@ import * as fsevents from "fsevents";
 // Handle --readme flag early (before Commander parses)
 handleReadmeFlag(import.meta.url);
 
+if (process.platform !== "darwin") {
+    console.error("fsevents-profile requires macOS (uses fs_usage and /dev/fsevents).");
+    process.exit(1);
+}
```
[the remark this answers is not in the document]
Comment (genesiscz) [reply src/fsevents-profile/index.ts:15]: @coderabbitai Fixed — replaced console.error with logger.error and removed stray console.log debug statements. [018ad30](https://github.com/genesiscz/GenesisTools/commit/018ad30e).

File: src/automate/lib/daemon.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (genesiscz) [reply src/automate/lib/daemon.ts]: @coderabbitai Fixed — removed the obvious comments per the no-obvious-comments guideline. [81237bc](https://github.com/genesiscz/GenesisTools/commit/81237bc85).

File: src/daemon/daemon.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (genesiscz) [reply src/daemon/daemon.ts]: @coderabbitai Fixed — removed the obvious comments per the no-obvious-comments guideline. [81237bc](https://github.com/genesiscz/GenesisTools/commit/81237bc85).
